.github/actions/setup-runner/action.yml

```diff
@@ -4,46 +4,58 @@ name: Set up Runner
 inputs:
   os:
     type: string
-    description: 'The operating system of the runner'
+    description: "The operating system of the runner"
     required: true
   gcc:
     type: string
-    description: 'The GCC version to use'
+    description: "The GCC version to use"
+    required: true
+  python-version:
+    type: string
+    description: "The python version to use"
     required: true
 
 runs:
   using: 'composite'
   steps:
-    - name: Install build commands, GCC and libxml2 (Linux)
-      if: ${{ startsWith(inputs.os,'ubuntu-') }}
+    # ------------------------------ Ubuntu ------------------------------
+    - if: ${{ startsWith(inputs.os,'ubuntu-') && (inputs.gcc != '')}}
+      run: sudo apt update
+      shell: bash
+    - if: ${{ startsWith(inputs.os,'ubuntu-') && (inputs.gcc != '')}}
+      run: sudo apt-get install -y make ninja-build ${{ inputs.gcc }} $(echo ${{ inputs.gcc }} | sed -e 's/gcc/g\+\+/') libxml2-utils
       shell: bash
-      run: |
-        sudo apt update
-        sudo apt-get install -y \
-          make \
-          ninja-build \
-          ${{ inputs.gcc }} \
-          $(echo ${{ inputs.gcc }} | sed -e 's/gcc/g\+\+/') \
-          libxml2-utils
 
-    - name: Install msys with GCC (Windows)
-      if: ${{ startsWith(inputs.os,'windows-') }}
+    # ------------------------------ Windows ------------------------------
+    - if: ${{ startsWith(inputs.os,'windows-') }}
       uses: msys2/setup-msys2@ddf331adaebd714795f1042345e6ca57bd66cea8
       with:
         install: gcc make
         cache: true
 
-    - name: Install ninja (Windows)
-      if: ${{ startsWith(inputs.os,'windows-') }}
+    # ------------------------------ MacOs ------------------------------
+    - if: ${{ startsWith(inputs.os,'macos-') }}
+      run: HOMEBREW_NO_AUTO_UPDATE=1 brew update
+      shell: bash
+    - if: ${{ startsWith(inputs.os,'macos-') }}
+      run: HOMEBREW_NO_AUTO_UPDATE=1 brew install libxml2
       shell: bash
-      run: |
-        choco install ninja
 
-    - name: Install ninja and libxml2 (MacOs)
-      if: ${{ startsWith(inputs.os,'macos-') }}
+    # ------------------------------ Python, all OS ------------------------------
+    - uses: actions/setup-python@v5
+      with:
+        python-version: ${{ inputs.python-version }}
+    - id: pip-cache
```

Comment:
FYI, I made an action that is a bit more clever about caching and is ABI-aware: https://github.com/marketplace/actions/cache-python-deps.

Reply:
For now I would keep it as it is.